### PR TITLE
Convert primary user feedback to cli

### PR DESCRIPTION
**Pull request by hadley**
These are the major bits of feedback generated by roxygen2. I'll do another pass on the smaller bits afterwards, but this already required a decent amount of refactoring so it's at a good place to review.

### CODE REVIEW

File: R/roxygenize-setup.R [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      cli::cli_rule()
+      cli::cli_inform(c(
+        "Changes in roxygen2 7.0.0:",
+        "* `%` is now escaped automatically in Markdown mode.",
+        "Please carefully check .Rd files for changes"
+      ))
+      cli::cli_rule()
+    }
+
+    cli::cli_inform("Setting {.var RoxygenNote} to {cur}")
```
Comment (gaborcsardi) [inline R/roxygenize-setup.R:49]: FWIW this was `{.str {cur}}` above.

File: R/roxygenize.R [hunk 2]
```diff
@@ -30,16 +30,10 @@ roxygenize <- function(package.dir = ".",
                        load_code = NULL,
                        clean = FALSE) {
 
-  base_path <- normalizePath(package.dir, mustWork = TRUE)
```
Comment (gaborcsardi) [inline R/roxygenize.R:33]: I wonder if we should still call `normalizePath()`?

**Review by gaborcsardi — APPROVED** [review]
Looks good. I had two small comments.

File: R/roxygenize.R (hunk 2 above)
Comment (hadley) [reply R/roxygenize.R:33]: What are the advantages? Keeping a relative path makes the paths in feedback much more useful.

Comment (gaborcsardi) [reply R/roxygenize.R:33]: At least on Windows `normalizePath()` does various conversions, so if we don't call it now, then the behaviour probably changes in some cases.

OTOH if it is only used for reporting then that is fine.

Comment (hadley) [reply R/roxygenize.R:33]: Hmmm, if I re-add it, none of the tests fail, so I must've only needed it in an interim state. Seems better to keep to minimise unintended consequences.

**Pull request merged**